eta/core/c3d.py

```diff
@@ -0,0 +1,340 @@
+                tf.Session instance is created, and you are responsible for
+                scalling the close() method of this class when you are done
+                computing
+            imgs: an optional tf.placeholder of size [XXXX, 16, 112, 112, 3] to
+                use. By default, a placeholder of size [None, 16, 112, 112, 3] is
+                used so you can evaluate any number of images at once
+        '''
+        self.config = config or C3DConfig.default()
+        self.sess = sess or tf.Session()
+        self.imgs = imgs or tf.placeholder(tf.float32, [None, 16, 112, 112, 3])
+        with tf.variable_scope('var_name') as var_scope:
+            weights = {
+                'wc1': self._variable_with_weight_decay('wc1', [3, 3, 3, 3, 64], 0.04, 0.00),
+                'wc2': self._variable_with_weight_decay('wc2', [3, 3, 3, 64, 128], 0.04, 0.00),
+                'wc3a': self._variable_with_weight_decay('wc3a', [3, 3, 3, 128, 256], 0.04, 0.00),
+                'wc3b': self._variable_with_weight_decay('wc3b', [3, 3, 3, 256, 256], 0.04, 0.00),
+                'wc4a': self._variable_with_weight_decay('wc4a', [3, 3, 3, 256, 512], 0.04, 0.00),
+                'wc4b': self._variable_with_weight_decay('wc4b', [3, 3, 3, 512, 512], 0.04, 0.00),
+                'wc5a': self._variable_with_weight_decay('wc5a', [3, 3, 3, 512, 512], 0.04, 0.00),
+                'wc5b': self._variable_with_weight_decay('wc5b', [3, 3, 3, 512, 512], 0.04, 0.00),
+                'wd1': self._variable_with_weight_decay('wd1', [8192, 4096], 0.04, 0.001),
+                'wd2': self._variable_with_weight_decay('wd2', [4096, 4096], 0.04, 0.002),
+                'out': self._variable_with_weight_decay('wout', [4096, 101], 0.04, 0.005)
+                }
+            biases = {
+                'bc1': self._variable_with_weight_decay('bc1', [64], 0.04, 0.0),
+                'bc2': self._variable_with_weight_decay('bc2', [128], 0.04, 0.0),
+                'bc3a': self._variable_with_weight_decay('bc3a', [256], 0.04, 0.0),
+                'bc3b': self._variable_with_weight_decay('bc3b', [256], 0.04, 0.0),
+                'bc4a': self._variable_with_weight_decay('bc4a', [512], 0.04, 0.0),
+                'bc4b': self._variable_with_weight_decay('bc4b', [512], 0.04, 0.0),
+                'bc5a': self._variable_with_weight_decay('bc5a', [512], 0.04, 0.0),
+                'bc5b': self._variable_with_weight_decay('bc5b', [512], 0.04, 0.0),
+                'bd1': self._variable_with_weight_decay('bd1', [4096], 0.04, 0.0),
+                'bd2': self._variable_with_weight_decay('bd2', [4096], 0.04, 0.0),
+                'out': self._variable_with_weight_decay('bout', [101], 0.04, 0.0),
+                }
+        self.build_c3d(self.imgs, self.config.dropout, self.config.batchsize,
+            weights, biases)
+        self._load_model()
+
+    def _variable_on_cpu(self, name, shape, initializer):
+        with tf.device('/cpu:0'):
```

Comment:
You are forcing CPU 0 here, but we want to use GPU! And we shouldn't be forcing any specific devices.

See `vgg16.py`. No devices are forced.

Reply:
Addressed.